### PR TITLE
Fixed TypeScript errors

### LINKED ISSUE #20: Fix TypeScript errors
When running lint check eslint throws errors. Need to fix this to prevent our GitHub actions from failing.

### PR DESCRIPTION
**Pull request by yesyash**
### Description 
Running `eslint` was throwing errors due to various rules defined in eslint config not being met. I have fixed most of the errors and added a Todo comment on the ones I was not able to fix and were of low priority.

### CODE REVIEW

File: routes/users.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import authenticate from "../middlewares/authenticate";
 
 const router = Router();
 
+// eslint-disable-next-line
 router.get("/self", authenticate, getSelfData);
```
Comment (harshith-venkatesh) [inline routes/users.ts:8]: Could we add a todo to check why we are adding this and how to mitigate that

**Review by harshith-venkatesh — APPROVED** [review]

**Review by harshith-venkatesh — APPROVED** [review]

**Comment by harshith-venkatesh** [comment]
Builds are failing due to logger and config
Are we looking into that?

File: @types/index.d.ts [hunk 2]
```diff
@@ -2,6 +2,6 @@ import config from "config";
 import winston from "winston";
 
 declare global {
-  var config: config;
-  var logger: winston.Logger;
+  let config: config;
```
Comment (ankurnarkhede) [inline @types/index.d.ts:5]: This has to be kept as `var`, to let config and logger be accessible globally. The CI is failing due to this.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: routes/users.ts (hunk 1 above)
Comment (yesyash) [reply routes/users.ts:13]: Thanks for mentioning this @harshith-venkatesh add a Todo comment.

**Review by ankurnarkhede — APPROVED** [review]

**Pull request merged**